lib/src/main/java/uk/gov/di/ipv/cri/address/library/service/AddressService.java

```diff
@@ -116,6 +118,33 @@ public void setAddressValidity(List<CanonicalAddress> addresses)
         }
     }
 
+    public List<Address> mapCanonicalAddresses(List<CanonicalAddress> addresses) {
+        return addresses.stream()
+                .map(
+                        address -> {
+                            Address mappedAddress = new Address();
+                            mappedAddress.setAddressCountry(address.getAddressCountry());
+                            mappedAddress.setAddressLocality(address.getAddressLocality());
+                            mappedAddress.setBuildingName(address.getBuildingName());
+                            mappedAddress.setBuildingNumber(address.getBuildingNumber());
+                            mappedAddress.setDepartmentName(address.getDepartmentName());
+                            mappedAddress.setDependentAddressLocality(
+                                    address.getDependentAddressLocality());
+                            mappedAddress.setDependentStreetName(address.getDependentStreetName());
+                            mappedAddress.setDoubleDependentAddressLocality(
+                                    address.getDoubleDependentAddressLocality());
+                            mappedAddress.setOrganisationName(address.getOrganisationName());
+                            mappedAddress.setPostalCode(address.getPostalCode());
+                            mappedAddress.setStreetName(address.getStreetName());
+                            mappedAddress.setSubBuildingName(address.getSubBuildingName());
+                            mappedAddress.setUprn(address.getUprn());
+                            mappedAddress.setValidFrom(address.getValidFrom());
+                            mappedAddress.setValidUntil(address.getValidUntil());
+                            return mappedAddress;
```

Comment:
Can we add a constructor to `Address` that takes a `CanonnicalAddress` and initialises all the fields as here?
Then the stream could be simply `.map(address -> new Address(address))`